src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +472,30 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str) -> str:
+        temp_stage = self._session.get_session_stage()
+        stage_path = path
+        if not path.startswith(STAGE_PREFIX) and not os.path.exists(path):
+            raise ValueError(f"The local file {path} does not exist")
+        else:
```

Comment:
Nit: no need to have `else` since the `if` is to validate input.

Reply:
revised